src/mca/gds/ds12/gds_dstore.c

```diff
@@ -2304,10 +2304,6 @@ inline pmix_status_t _dstore_fetch(const char *nspace, pmix_rank_t rank, const c
         return rc;
     }
 
-    if (kvs) {
-        *kvs = NULL;
```

Comment:
This doesn't sound right to me. it doesn't address NULL-pointer dereference error.
The problem is in the case where `kvs == NULL` (which is assumed by this if) and not with `*kvs == NULL`.
Instead of removing this you need to require `kvs != NULL` and error exit if this is not true. 
It doesn't make sense to call this function if you don't want to receive the answer.